### ISSUE TITLE
cython implementation

### ISSUE THREAD

**Issue by lguirola** [lguirola/sta663-Final-Project#9]
@shldngzh 

Hey, so I've spent last night working on this. I have for now two problems:
1. I need to use cython stuff to deal with the inner loop
2. There is some problem with feeding the function, which throws an error.

Do you think you could have a look when you have the chance? Otherwise I will keep working on it

**Comment by xzhai12** [comment]
@lguirola 

Yeah definitely. I would look at it after today - I think I can ignore armadillo and do the matrix inner product and multiplication things by ourself, so no errors will be raised and I still expect running efficiency for large and sparse matrix. 

Maybe tomorrow I will commit my new version of coordinate descent and some comments on your nice work. Of course if you could continue to work on it, it will be great - but if you do not have time you can of course take care of other things first since the dead line is like April 26, when we can definitely finish everything.

**Comment by lguirola** [comment]
@shldngzh 

I've looked at the job you did with c++ and it looks amazing! I really want to thank you for it. 

I've tried to make progress on this during the last two days and a half, but I've been unsucessful :(, it's being pretty frustrating. Can we have a look at it together or perhaps you try to have a look at it? 

Also, we need to coodinate on what to do once done with the optimization step, I can work on that if you want.
